Snowflake.Data/Core/Authenticator/OktaAuthenticator.cs

```diff
@@ -67,7 +67,7 @@ async Task IAuthenticator.AuthenticateAsync(CancellationToken cancellationToken)
             logger.Debug("step 4: get SAML reponse from sso");
             var samlRestRequest = BuildSAMLRestRequest(ssoUrl, onetimeToken);
             var samlRawResponse = await session.restRequester.GetAsync(samlRestRequest, cancellationToken);
-            var samlRawHtmlString = samlRawResponse.Content.ReadAsStringAsync().Result;
+            var samlRawHtmlString = await samlRawResponse.Content.ReadAsStringAsync();
```

Comment:
Let me understand about `await`. So this will spawn a thread to run ReadAsStringAsync() and the main thread will wait for it to finish in line 73?

Reply:
Not necessary spawning a thread right here. The compiler would rewrite the code here to be a state machine, and decide whether it should proceed or return. This is the same pattern as asynchronous programming, but .NET allow you to yield the ugly work to compiler using async and await keywords.

For further reading: https://devblogs.microsoft.com/premier-developer/dissecting-the-async-methods-in-c/